### PR TITLE
chore!: Hard-deprecate `as_adjacency_matrix(edges = )`

### PR DESCRIPTION
**Pull request by krlmlr**
For #815, more work needed. Running revdepchecks now.

### CODE REVIEW

**Comment by krlmlr** [comment]
@maelle: This also affects `[.igraph(edges = )` .  Can you please take over here?

**Comment by krlmlr** [comment]
See the failing tests with this PR.

File: R/conversion.R [hunk 1]
```diff
@@ -346,14 +346,14 @@ as_adjacency_matrix <- function(graph, type = c("both", "upper", "lower"),
...
-  if (!missing(edges)) {
-    warning("The `edges` argument of `as_adjacency_matrix` is deprecated; it will be removed in igraph 1.4.0")
+  if (!missing(edges) && isTRUE(edges)) {
+    lifecycle::deprecate_stop("2.0.0", "as_adjacency_matrix(edges = )")
   }
 
-  if (!sparse) {
-    get.adjacency.dense(graph, type = type, attr = attr, weights = NULL, names = names)
-  } else {
+  if (sparse) {
```
Comment (maelle) [inline R/conversion.R:353]: why this change?

Comment (krlmlr) [reply R/conversion.R:353]: Reduces cyclomatic complexity.

Comment (krlmlr) [reply R/conversion.R:353]: Or not? Anyway, easier to read, and I was touching the code next to it.

Could have been two different PRs, but I chose to use just one.

**Comment by maelle** [comment]
@krlmlr 
- will the "!" in the commit message (from the current PR title) be enough to ensure this goes into the changelog as breaking change? 
- can you please run revdeps checks for this change?

File: R/conversion.R [hunk 2]
```diff
@@ -344,14 +344,14 @@ as_adjacency_matrix <- function(graph, type = c("both", "upper", "lower"),
...
     get.adjacency.sparse(graph, type = type, attr = attr, edges = edges, names = names)
```
Comment (krlmlr) [inline R/conversion.R:352]: This now looks like dead code.

File: R/conversion.R [hunk 3]
```diff
@@ -317,7 +317,7 @@ get.adjacency.sparse <- function(graph, type = c("both", "upper", "lower"),
 #'   numeric. If the `sparse` argument is `FALSE`, then character is
 #'   also allowed. The reason for the difference is that the `Matrix`
 #'   package does not support character sparse matrices yet.
-#' @param edges Logical scalar, whether to return the edge ids in the matrix.
+#' @param edges `r lifecycle::badge("deprecated")` Logical scalar, whether to return the edge ids in the matrix.
 #'   For non-existant edges zero is returned.
```
Comment (krlmlr) [inline R/conversion.R:321]: Should we just mention that this must be omitted?

File: R/conversion.R [hunk 4]
```diff
@@ -344,14 +344,14 @@ as_adjacency_matrix <- function(graph, type = c("both", "upper", "lower"),
                                 sparse = igraph_opt("sparsematrices")) {
   ensure_igraph(graph)
 
-  if (!missing(edges)) {
-    warning("The `edges` argument of `as_adjacency_matrix` is deprecated; it will be removed in igraph 1.4.0")
+  if (!missing(edges) && isTRUE(edges)) {
+    lifecycle::deprecate_stop("2.0.0", "as_adjacency_matrix(edges = )")
   }
```
Comment (krlmlr) [inline R/conversion.R:349]: Add a `deprecate_soft()` if `edges` is `FALSE` ?

**Review by krlmlr — COMMENTED** [review]
I'll run revdepchecks.

**Comment by krlmlr** [comment]
revdepchecks look fine.
